### PR TITLE
chore(site): refactor starter templates to use react-query

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

**Review by code-asher — APPROVED** [review]
Nice!  I like how this simplifies the story mocks, too.  Love the commits too.

File: site/src/pages/StarterTemplatePage/StarterTemplatePage.tsx [hunk 1]
```diff
@@ -1,31 +1,30 @@
...
 const StarterTemplatePage: FC = () => {
   const { exampleId } = useParams() as { exampleId: string };
   const organizationId = useOrganizationId();
-  const [state] = useMachine(starterTemplateMachine, {
-    context: {
-      organizationId,
-      exampleId,
-    },
-  });
+  const templateExamplesQuery = useQuery(templateExamples(organizationId));
```
Comment (code-asher) [inline site/src/pages/StarterTemplatePage/StarterTemplatePage.tsx:13]: Not something to do in this PR, but just a note that maybe we need a way to get a template by ID, seems unfortunate to query all the templates just to get one!

Comment (BrunoQuaresma) [reply site/src/pages/StarterTemplatePage/StarterTemplatePage.tsx:13]: I do agree with you!

**Comment by BrunoQuaresma** [comment]
Related to https://github.com/coder/coder/issues/9598

**Pull request merged**